tests/store/model_registry/test_sqlalchemy_store.py

```diff
@@ -125,30 +125,83 @@ def test_delete_registered_model(self):
-        self.assertEqual(set([rm.name for rm in self.store.list_registered_models()]),
+        self.assertEqual(set(self._list_registered_models()),
                          set(["A", "B", "BB", "BA", "AB", "BBC"]))
 
         # list should not return deleted models
         self.store.delete_registered_model(name="BA")
         self.store.delete_registered_model(name="B")
-        self.assertEqual(set([rm.name for rm in self.store.list_registered_models()]),
+        self.assertEqual(set(self._list_registered_models()),
                          set(["A", "BB", "AB", "BBC"]))
 
+    def test_list_registered_model_paginated(self):
+        rms = [self._rm_maker(f"RM{i:03}").name for i in range(50)]
+
+        # test flow with fixed max_results
+        returned_rms = []
+        result = self._list_registered_models(page_token=None, max_results=5)
+        returned_rms.extend(result)
+        while result.token:
+            result = self._list_registered_models(page_token=result.token, max_results=5)
+            returned_rms.extend(result)
+        self.assertEqual(set(rms), set(returned_rms))
+
+        # test that pagination will return all valid results in sorted order
+        # by name ascending
+        result = self._list_registered_models(max_results=5)
+        self.assertNotEqual(result.token, None)
+        self.assertEqual(result, rms[0:5])
+
+        result = self._list_registered_models(page_token=result.token, max_results=10)
+        self.assertNotEqual(result.token, None)
+        self.assertEqual(result, rms[5:15])
+
+        result = self._list_registered_models(page_token=result.token, max_results=20)
+        self.assertNotEqual(result.token, None)
+        self.assertEqual(result, rms[15:35])
+
+        result = self._list_registered_models(page_token=result.token, max_results=100)
+        # assert that page token is None
+        self.assertEqual(result.token, None)
+        self.assertEqual(result, rms[35:])
+
+        # test that providing a completely invalid page token throws
+        with self.assertRaises(MlflowException) as exception_context:
+            self._list_registered_models(page_token="evilhax", max_results=20)
+        assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+
+        # test that providing too large of a max_results throws
+        with self.assertRaises(MlflowException) as exception_context:
+            self._list_registered_models(page_token="evilhax", max_results=1e15)
+            assert exception_context.exception.error_code == ErrorCode.Name(INVALID_PARAMETER_VALUE)
+        self.assertIn("Invalid value for request parameter max_results",
+                      exception_context.exception.message)
+        # list should not return deleted models
+        self.store.delete_registered_model(name=f"RM{0:03}")
+        self.assertEqual(set(self._list_registered_models(max_results=100)),
+                         set(rms[1:]))
```

Comment:
Nit: I'd prefer that a test function test only one scenario rather than have one test function validate multiple scenarios.